src/librustc/ich/impls_syntax.rs

```diff
@@ -43,25 +43,23 @@ impl<'a> HashStable<StableHashingContext<'a>> for [ast::Attribute] {
     }
 }
 
-impl<'a> HashStable<StableHashingContext<'a>> for ast::Attribute {
-    fn hash_stable(&self, hcx: &mut StableHashingContext<'a>, hasher: &mut StableHasher) {
+impl<'ctx> syntax::HashStableContext for StableHashingContext<'ctx> {
+    fn hash_attr(&mut self, attr: &ast::Attribute, hasher: &mut StableHasher) {
```

Comment:
I think it would be better to have a `is_ignored_attr` method on `syntax::HashStableContext` and move the impl to `syntax`.

Reply:
This doesn't work out well unfortunately. I'm only able to move one of the impls, but not the `impl<AstCtx: crate::HashStableContext> HashStable<AstCtx> for [ast::Attribute] {` one because it becomes orphaned (including with specialization). So I would prefer to have those impls in one place and this file is the only way to do that I think.